messages/src/lib.rs

```diff
@@ -140,39 +138,93 @@ impl TryFrom<&Url> for url::Url {
     }
 }
 
-/// DAP protocol message representing a duration with a resolution of seconds.
+/// DAP protocol message representing a duration as a multiple of the task's time precision.
+/// The value represents the number of time_precision intervals.
+///
+/// To convert between this representation and real-world durations (seconds),
+/// use the conversion methods that take a [`TimePrecision`] parameter.
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
 pub struct Duration(u64);
 
 impl Duration {
-    pub const ZERO: Duration = Duration::from_seconds(0);
+    pub const ZERO: Duration = Duration::from_time_precision_units(0);
+    pub const ONE: Duration = Duration::from_time_precision_units(1);
 
-    /// Create a duration representing the provided number of seconds.
-    pub const fn from_seconds(seconds: u64) -> Self {
-        Self(seconds)
+    /// Create a duration representing the provided number of seconds, given the task's time precision.
+    ///
+    /// The duration will be rounded down to the nearest multiple of time_precision.
+    ///
+    /// # Arguments
+    /// * `seconds` - Duration in seconds
+    /// * `time_precision` - The task's time precision
+    ///
+    /// # Panics
+    /// Panics if `time_precision.as_seconds()` is 0.
+    ///
```

Comment:
nit: this is the style we use elsewhere. 
```suggestion
    /// # Arguments
    ///
    /// * `seconds` - Duration in seconds
    /// * `time_precision` - The task's time precision
    ///
    /// # Panics
    ///
    /// Panics if `time_precision.as_seconds()` is 0.
```